annotate.el

```diff
@@ -115,6 +115,43 @@
   :type 'string
   :group 'annotate)
 
+(defcustom annotate-maximum-size-checksum 5000000
+  "Calculate checksum of  the current buffer only if  the size is
+less than this size (in characters)"
+  :type  'integer
+  :group 'annotate)
+
+(defconst annotate-warn-file-changed-control-string
+  (concat "The file '%s' has changed on disk "
+          "from the last time the annotations was saved. "
+          "Chances are that they will not be displayed correctly")
+  "The  message to  warn user  that  file has  been modified  and
+  annototatnions position could be outdated")
+
+(defcustom annotate-blacklist-major-mode '(org-mode)
+  "Prevent  loading of  annotated-mode  When  the visited  file's
+major mode is a member of this list (space separated entries)."
+  :type  '(repeat symbol)
+  :group 'annotate)
+
+(defun annotate-after-hook ()
+  (let ((annotate-allowed-p (with-current-buffer (current-buffer)
+                              (not (cl-member major-mode annotate-blacklist-major-mode)))))
+    (cond
+     ((not annotate-allowed-p)
+      (annotate-shutdown)
+      (setq annotate-mode nil))
+    (annotate-mode
+     (annotate-initialize))
+    (t
+     (annotate-shutdown)))))
+
+(cl-defun annotate-buffer-checksum (&optional (object (current-buffer)))
+  (if (< (buffer-size)
+         annotate-maximum-size-checksum)
+      (md5 object)
+    nil))
```

Comment:
Does annotate.el already require `cl`? If not, could this be implemented without `cl`?

Reply:
Please add a docstring.

Reply:
IIRC cl-defun is needed because of the default optional argument

https://www.gnu.org/software/emacs/manual/html_mono/cl.html#Argument-Lists

So maybe the right question is: "is an optional argument needed at all?" :)

Reply:
Silly me. I didn't even realize that annotate.el *already* requires `cl`. So this line of inquiry was pointless. Sorry about that.